spec/koans.js

```diff
@@ -86,40 +86,48 @@ describe('`string.includes()` finds string within another string. ', () => {
   describe('find a single character', function() {
     it('in a three char string', function() {
       /* const searchString = ???? */
-      //expect('xyz'.includes(searchString)).toBe(true);
+      const searchString =  "x"   //.includes("xyz");
+      expect('xyz'.includes(searchString)).toBe(true);
     });
     it('reports false if character was not found', function() {
       /* const expected = ????*/;
-      //expect('xyz'.includes('abc')).toBe(expected);
+      const expected = false
+      expect('xyz'.includes('abc')).toBe(expected);
     });
   });
 
   describe('find a string', function() {
     it('that matches exactly', function() {
+      const findSome = parameter => 'xyz'.includes();
       /* const findSome = .... => 'xyz'.includes();*/
-      //expect(findSome('xyz')).toBe(true);
+        // function findSome(par) {
+        // return "xyz".includes(par)
+        //   }
+      
+      const findSome = sentence.includes('z')
+      expect(findSome('xyz')).toBe(true);
     });
   });
 
   describe('search for an empty string, is always true', function() {
     it('in an empty string', function() {
-      /* .... */
-      //expect(''.includes(x)).toBe(true);
+      /* .... */ const x = ''
+      expect(''.includes(x)).toBe(true);
     });
     it('in `abc`', function() {
-      /* .... */
-      //expect('abc'.includes(x)).toBe(true);
+      /* .... */const x ='abc'
+      expect('abc'.includes(x)).toBe(true);
     });
   });
 
   describe('takes a position from where to start searching', function() {
     it('does not find `a` after position 1 in `abc`', function() {
       /*....*/
-      //expect('abc'.includes('a', position)).toBe(false);
+      expect('abc'.includes('a', position)).toBe(false);
     });
     it('even the position gets coerced', function() {
       /*const findAtPosition = (pos) => 'xyz'.includes(?????);*/
-      //expect(findAtPosition('2')).toBe(true);
+      expect(findAtPosition('2')).toBe(true);
     });
     describe('invalid positions get converted to 0', function() {
```

Comment:
This section is about understanding how to use the includes method with two arguments, if you use the includes method with 2 args, the first one will be your search string and the second one will be the position (the index) from where to start the search. Invalid positions, that is indexes that do not exist in the original string, get converted to 0.
```js
  
  describe('takes a position from where to start searching', function() {
    it('does not find `a` after position 1 in `abc`', function() {
      const position = 1;
      expect('abc'.includes('a', position)).toBe(false);
    });
    it('even the position gets coerced', function() {
      const findAtPosition = (pos) => 'xyz'.includes('z', pos); 
      expect(findAtPosition('2')).toBe(true);
    });
    describe('invalid positions get converted to 0', function() {
      it('e.g. `undefined`', function() {
        const findAtPosition = (pos) => 'xyz'.includes('x', pos); 
        expect(findAtPosition(void 0)).toBe(true);
      });
      it('negative numbers', function() {
        const findAtPosition = (pos) => 'xyz'.includes('x', pos); 
        expect(findAtPosition(-2)).toBe(true);
      });
      it('NaN', function() {
        const findAtPosition = (pos) => 'xyz'.includes('x', pos); 
        expect(findAtPosition(NaN)).toBe(true);
      });
    });
  });

```